File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Core/AlibabaCloud/SDK/ROS/CDK/Core/ValidationResults.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Core
{
    /// <summary>A collection of validation results.</summary>
    [JsiiClass(nativeType: typeof(AlibabaCloud.SDK.ROS.CDK.Core.ValidationResults), fullyQualifiedName: "@alicloud/ros-cdk-core.ValidationResults", parametersJson: "[{\"name\":\"results\",\"optional\":true,\"type\":{\"collection\":{\"elementtype\":{\"fqn\":\"@alicloud/ros-cdk-core.ValidationResult\"},\"kind\":\"array\"}}}]")]
    public class ValidationResults : DeputyBase
    {
        public ValidationResults(AlibabaCloud.SDK.ROS.CDK.Core.ValidationResult[]? results = null): base(new DeputyProps(new object?[]{results}))
        {
        }

        /// <summary>Used by jsii to construct an instance of this class from a Javascript-owned object reference</summary>
        /// <param name="reference">The Javascript-owned object reference</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected ValidationResults(ByRefValue reference): base(reference)
        {
        }

        /// <summary>Used by jsii to construct an instance of this class from DeputyProps</summary>
        /// <param name="props">The deputy props</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected ValidationResults(DeputyProps props): base(props)
        {
        }

        [JsiiMethod(name: "collect", parametersJson: "[{\"name\":\"result\",\"type\":{\"fqn\":\"@alicloud/ros-cdk-core.ValidationResult\"}}]")]
        public virtual void Collect(AlibabaCloud.SDK.ROS.CDK.Core.ValidationResult result)
        {
            InvokeInstanceVoidMethod(new System.Type[]{typeof(AlibabaCloud.SDK.ROS.CDK.Core.ValidationResult)}, new object[]{result});
        }

        [JsiiMethod(name: "errorTreeList", returnsJson: "{\"type\":{\"primitive\":\"string\"}}")]
        public virtual string ErrorTreeList()
        {
            return InvokeInstanceMethod<string>(new System.Type[]{}, new object[]{})!;
        }

        /// <summary>Wrap up all validation results into a single tree node.</summary>
        /// <remarks>
        /// If there are failures in the collection, add a message, otherwise
        /// return a success.
        /// </remarks>
        [JsiiMethod(name: "wrap", returnsJson: "{\"type\":{\"fqn\":\"@alicloud/ros-cdk-core.ValidationResult\"}}", parametersJson: "[{\"name\":\"message\",\"type\":{\"primitive\":\"string\"}}]")]
        public virtual AlibabaCloud.SDK.ROS.CDK.Core.ValidationResult Wrap(string message)
        {
            return InvokeInstanceMethod<AlibabaCloud.SDK.ROS.CDK.Core.ValidationResult>(new System.Type[]{typeof(string)}, new object[]{message})!;
        }

        [JsiiProperty(name: "isSuccess", typeJson: "{\"primitive\":\"boolean\"}")]
        public virtual bool IsSuccess
        {
            get => GetInstanceProperty<bool>()!;
        }

        [JsiiProperty(name: "results", typeJson: "{\"collection\":{\"elementtype\":{\"fqn\":\"@alicloud/ros-cdk-core.ValidationResult\"},\"kind\":\"array\"}}")]
        public virtual AlibabaCloud.SDK.ROS.CDK.Core.ValidationResult[] Results
        {
            get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ValidationResult[]>()!;
            set => SetInstanceProperty(value);
        }
    }
}

File: multiple-languages/go/alicloudroscdkcore/ValidationResults.go
package alicloudroscdkcore

import (
	_init_ "github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore/jsii"
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"
)

// A collection of validation results.
type ValidationResults interface {
	IsSuccess() *bool
	Results() *[]ValidationResult
	SetResults(val *[]ValidationResult)
	Collect(result ValidationResult)
	ErrorTreeList() *string
	// Wrap up all validation results into a single tree node.
	//
	// If there are failures in the collection, add a message, otherwise
	// return a success.
	Wrap(message *string) ValidationResult
}

// The jsii proxy struct for ValidationResults
type jsiiProxy_ValidationResults struct {
	_ byte // padding
}

func (j *jsiiProxy_ValidationResults) IsSuccess() *bool {
	var returns *bool
	_jsii_.Get(
		j,
		"isSuccess",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_ValidationResults) Results() *[]ValidationResult {
	var returns *[]ValidationResult
	_jsii_.Get(
		j,
		"results",
		&returns,
	)
	return returns
}


func NewValidationResults(results *[]ValidationResult) ValidationResults {
	_init_.Initialize()

	j := jsiiProxy_ValidationResults{}

	_jsii_.Create(
		"@alicloud/ros-cdk-core.ValidationResults",
		[]interface{}{results},
		&j,
	)

	return &j
}

func NewValidationResults_Override(v ValidationResults, results *[]ValidationResult) {
	_init_.Initialize()

	_jsii_.Create(
		"@alicloud/ros-cdk-core.ValidationResults",
		[]interface{}{results},
		v,
	)
}

func (j *jsiiProxy_ValidationResults)SetResults(val *[]ValidationResult) {
	if err := j.validateSetResultsParameters(val); err != nil {
		panic(err)
	}
	_jsii_.Set(
		j,
		"results",
		val,
	)
}

func (v *jsiiProxy_ValidationResults) Collect(result ValidationResult) {
	if err := v.validateCollectParameters(result); err != nil {
		panic(err)
	}
	_jsii_.InvokeVoid(
		v,
		"collect",
		[]interface{}{result},
	)
}

func (v *jsiiProxy_ValidationResults) ErrorTreeList() *string {
	var returns *string

	_jsii_.Invoke(
		v,
		"errorTreeList",
		nil, // no parameters
		&returns,
	)

	return returns
}

func (v *jsiiProxy_ValidationResults) Wrap(message *string) ValidationResult {
	if err := v.validateWrapParameters(message); err != nil {
		panic(err)
	}
	var returns ValidationResult

	_jsii_.Invoke(
		v,
		"wrap",
		[]interface{}{message},
		&returns,
	)

	return returns
}

